text/0002-sarif-result-schema.md

```diff
@@ -0,0 +1,1939 @@
+    {
+      "name": "reduce-outdated-minor-dependencies",
+      "summary": "Update outdated minor versions",
+      "details": "9 minor versions outdated",
+      "defaultThreshold": 0.05,
+      "items": [],
+      "input": 0.0782608695652174
+    },
+    {
+      "name": "reduce-outdated-dependencies",
+      "summary": "Update outdated versions",
+      "details": "34% of versions outdated",
+      "defaultThreshold": 0.2,
+      "items": [],
+      "input": 0.3391304347826087
+    }
+  ]
+}
+```
+</details>
+
+Here is a full JSON output from running checkup _using SARIF_. I have removed most of the lists of file names to keep it clean, but it was 26487 lines of JSON.
+<details>
+
+  <summary>SARIF output</summary>
+
+  ```
+{
+  "version": "2.1.0",
+  "properties": {
+    "project": {
+      "name": "travis",
+      "version": "0.0.1",
+      "repository": {
+        "totalCommits": 5870,
+        "totalFiles": 1538,
+        "age": "8 years",
+        "activeDays": "1380",
+        "linesOfCode": {
+          "types": [
+            {
+              "total": 44939,
+              "extension": "js"
+            },
+            {
+              "total": 201,
+              "extension": "html"
+            },
+            {
+              "total": 14355,
+              "extension": "scss"
+            },
+            {
+              "total": 10803,
+              "extension": "hbs"
+            },
+            {
+              "total": 22836,
+              "extension": "svg"
+            },
+            {
+              "total": 608,
+              "extension": "rb"
+            }
+          ],
+          "total": 93742
+        }
+      }
+    },
+    "cli": {
+      "configHash": "1913a68bff82a711d2a1395e25d6992e",
+      "config": {
+        "plugins": [
+          "checkup-plugin-ember",
+          "checkup-plugin-ember-octane",
+          "checkup-plugin-javascript"
+        ],
+        "tasks": {
+          "ember/ember-test-types": [
+            "on",
+            {
+              "actions": {
+                "ratioApplicationTests": {
+                  "threshold": 3
+                }
+              }
+            }
+          ]
+        }
+      },
+      "version": "0.7.4",
+      "schema": 1,
+      "args": {
+        "paths": []
+      },
+      "flags": {
+        "format": "json",
+        "outputFile": ""
+      },
+      "timings": {
+        "ember/ember-types": 23.371086841,
+        "ember/ember-in-repo-addons-engines": 23.329550989,
+        "ember/ember-dependencies": 23.327852732,
+        "javascript/eslint-summary": 20.294677398,
+        "ember/ember-test-types": 23.346301169,
+        "ember-octane/ember-octane-migration-status": 8.863200936,
+        "ember/ember-template-lint-disables": 27.640971204,
+        "javascript/eslint-disables": 17.036571088,
+        "javascript/outdated-dependencies": 43.921809849
+      }
+    },
+    "analyzedFiles": [],
+    "analyzedFilesCount": 1539,
+    "actions": [
```

Comment:
Is there a more "SARIF"-style location for actions?

Reply:
So I don't think so. I have tried, specifically tried moving the actions to be stored with the Results because that makes more sense with SARIF's mental model, but I couldn't find anywhere to put them. Do you have any ideas?

Reply:
No, it was more of a question. I got nothing :(